node/src/components/consensus/era_supervisor.rs

```diff
@@ -113,6 +113,8 @@ pub struct EraSupervisor<I> {
     next_upgrade_activation_point: Option<ActivationPoint>,
     /// If true, the process should stop execution to allow an upgrade to proceed.
     stop_for_upgrade: bool,
+    /// Set to true when InitializeEras is handled.
+    is_initialized: bool,
```

Comment:
Can't we reuse `finished_joining`?

Reply:
I don't think so - the semantics are rather different. `finished_joining` kind of works the other way round, it's a way for the era supervisor to know that the reactor has transitioned. `is_initialized` is a way for the reactor to check if the era supervisor is ready.

Reply:
In my mind, `EraSupervisor` is "ready" _iff_ it's both initialized and finished joining.

Reply:
But "finished joining" is not a property of the era supervisor - it's a property of the node. The era supervisor just remembers if the node has this property in order to handle some things properly. `is_initialized`, on the other hand, is a property of the era supervisor.

Reply:
From the outside they're both the same – whether consensus component is ready to start participating in the active era.

Reply:
But conceptually they're not, and conflating things that are conceptually different tends to cause problems down the line.